### PR TITLE
Run go vet in CI

### PR DESCRIPTION
**Pull request by pdecat**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: test
-on: [push]
+on: [push, pull_request]
```
Comment (pdecat) [inline .github/workflows/test.yml:2]: Github Actions checks were not running for PRs without this: https://github.com/heimweh/go-pagerduty/pull/96/checks

**Comment by pdecat** [comment]
The failing check intentionally demonstrates the need for this PR, as the underlying issue caused by https://github.com/heimweh/go-pagerduty/pull/92 would have avoided a fix with a second PR (https://github.com/heimweh/go-pagerduty/pull/94).

**Comment by imjaroiswebdev** [comment]
Hi @pdecat could you please **rebase+push** this branch for me to proceed with the merge?

**Comment by pdecat** [comment]
Hi @imjaroiswebdev, done!

**Review by imjaroiswebdev — APPROVED** [review]
Thank you for your contribution. This will be very helpful @pdecat 💪🏽😎

**Pull request merged**